### PR TITLE
20.11.5.ug el8

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

File: slurm.spec [hunk 1]
```diff
@@ -67,12 +67,14 @@ Source:		%{slurm_source_dir}.tar.bz2
 %global _hardened_ldflags "-Wl,-z,lazy"
 
 Requires: munge
+Requires: json-c
```
Comment (stdweird) [inline slurm.spec:70]: this needs some distinguishing between el7 and el8

File: slurm-no-gpu.spec [hunk 2]
```diff
@@ -0,0 +1,679 @@
...
+%bcond_without pam
+%bcond_without x11
+
+# Disable hardened builds. -z,now or -z,relro breaks the plugin stack
+%undefine _hardened_build
+%global _hardened_cflags "-Wl,-z,lazy"
+%global _hardened_ldflags "-Wl,-z,lazy"
+
+Requires: munge
+Requires: json-c
```
Comment (stdweird) [inline slurm-no-gpu.spec:70]: needs distinguish between el7 and el8

File: slurm-no-gpu.spec [hunk 3]
```diff
@@ -0,0 +1,679 @@
...
+Slurm controller daemon. Used to manage the job queue, schedule jobs,
+and dispatch RPC messages to the slurmd processon the compute nodes
+to launch jobs.
+
+%package slurmd
+Summary: Slurm compute node daemon
+Group: System Environment/Base
+Requires: %{name}%{?_isa} = %{version}-%{release}
+%if %{with pmix} && "%{_with_pmix}" == "--with-pmix"
+Requires: pmix = %{pmix_version}
```
Comment (stdweird) [inline slurm-no-gpu.spec:232]: this is very odd. how does this become the slurm version...

File: bdist_rpm-no-gpu.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+# cuda-nvml-dev provides the nvml.h (put is part of the cuda rpms so has cud version in the name, hence the *)
+# however, it's not in the default include paths nor the slurm hardcoded ones, so use CPATH
+# nvml api is not really cuda specific, last API is from cuda 5, so any recent cuda will do
+
+# TODO: what if more than one cuda is available/installed, then the * thingies will probably not work
+# pmix-3 as rebuild from github src.rpm includes the devel rpms in the rpm
+sudo yum remove -y nvidia-driver-devel cuda-nvml-dev-10-2 nvidia-driver-NVML 
+sudo yum install -y ucx-devel "pmix-devel > 3.0.0" numactl-devel hwloc-devel
+
+# remove json-c12 -> use plain json-c
```
Comment (stdweird) [inline bdist_rpm-no-gpu.sh:34]: unless there is some hackery going on here, this is not ok for el7 i think

File: bdist_rpm-no-gpu.sh [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+# there's no option to pass nvml, it is only autodetected
+# nvidia-driver-devel provides the libnividia-ml.so symlnk, the real .so.1 comes from nvidia-driver-NVML
+# cuda-nvml-dev provides the nvml.h (put is part of the cuda rpms so has cud version in the name, hence the *)
+# however, it's not in the default include paths nor the slurm hardcoded ones, so use CPATH
+# nvml api is not really cuda specific, last API is from cuda 5, so any recent cuda will do
+
+# TODO: what if more than one cuda is available/installed, then the * thingies will probably not work
+# pmix-3 as rebuild from github src.rpm includes the devel rpms in the rpm
+sudo yum remove -y nvidia-driver-devel cuda-nvml-dev-10-2 nvidia-driver-NVML
```
Comment (stdweird) [inline bdist_rpm-no-gpu.sh:31]: also remove nvidia 10-1

File: bdist_rpm-no-gpu.sh [hunk 6]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
```
Comment (stdweird) [inline bdist_rpm-no-gpu.sh:1]: remove this altogether. the bdist_rpm does both

File: bdist_rpm.sh [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+# there's no option to pass nvml, it is only autodetected
+# nvidia-driver-devel provides the libnividia-ml.so symlnk, the real .so.1 comes from nvidia-driver-NVML
+# cuda-nvml-dev provides the nvml.h (put is part of the cuda rpms so has cud version in the name, hence the *)
+# however, it's not in the default include paths nor the slurm hardcoded ones, so use CPATH
+# nvml api is not really cuda specific, last API is from cuda 5, so any recent cuda will do
+
+# TODO: what if more than one cuda is available/installed, then the * thingies will probably not work
+# pmix-3 as rebuild from github src.rpm includes the devel rpms in the rpm
+sudo yum remove -y nvidia-driver-devel cuda-nvml-dev-10-2 nvidia-driver-NVML 
+sudo yum install -y ucx-devel "pmix-devel > 3.0.0" numactl-devel hwloc-devel
```
Comment (stdweird) [inline bdist_rpm.sh:73]: these should have already been installed

File: bdist_rpm.sh [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+# remove json-c12 -> use plain json-c
+sudo yum install -y json-c-devel json-c
```
Comment (stdweird) [inline bdist_rpm.sh:76]: should already be installed

File: bdist_rpm.sh [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+# also, specify python2 as python to require
+
+# glob expansion in list
```
Comment (stdweird) [inline bdist_rpm.sh:79]: this is for gpus
remove it, and the defines below that use nvmls variable

File: auxdir/x_ac_json.m4 [hunk 10]
```diff
@@ -33,11 +33,11 @@ AC_DEFUN([X_AC_JSON], [
         for d in $_x_ac_json_dirs; do
           test -d "$d" || continue
           test -d "$d/include" || continue
-          test -f "$d/include/json-c/json_object.h" || test -f "$d/include/json/json_object.h" || continue
+          test -f "$d/include/json-c12/json_object.h" || test -f "$d/include/json/json_object.h" || continue
```
Comment (stdweird) [inline auxdir/x_ac_json.m4:36]: does this work on el8? i don't think so

Comment (itkovian) [reply auxdir/x_ac_json.m4:36]: Hmm, should be the other way around. Also, I do _not_ recall changing anything ever in this file, but it did build.

File: bdist_rpm-no-gpu.sh (hunk 4 above)
Comment (itkovian) [reply bdist_rpm-no-gpu.sh:34]: Well, we won't be having it on EL7, or will we?

File: auxdir/x_ac_json.m4 (hunk 10 above)
Comment (stdweird) [reply auxdir/x_ac_json.m4:36]: this is correct for el7 (which requires josn-c12); but not on el8, where it is called json-c

File: bdist_rpm-no-gpu.sh (hunk 4 above)
Comment (stdweird) [reply bdist_rpm-no-gpu.sh:34]: we are not going to update  the el7 clusters?

File: slurm.spec (hunk 1 above)
Comment (stdweird) [reply slurm.spec:70]: yes. in both

File: auxdir/x_ac_json.m4 (hunk 10 above)
Comment (stdweird) [reply auxdir/x_ac_json.m4:36]: @itkovian ouch. this one still needs to be addressed at some point (but only relevant for burstbuffer and restd)

**Pull request merged**